packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -461,431 +511,744 @@ describe('Testing authentication module', () => {
-      expect(message).toBeNull();
-      expect(closeWindowSpy).not.toHaveBeenCalled();
-
-      utils.respondToMessage(initMessage, 'authentication');
-      await initPromise;
-      message = utils.findMessageByFunc('authentication.authenticate.failure');
-      expect(message).not.toBeNull();
-
-      // Wait 100ms for the message queue and 200ms for the close delay
-      await new Promise<void>(resolve =>
-        setTimeout(() => {
-          expect(closeWindowSpy).toHaveBeenCalled();
-          resolve();
-        }, 350),
-      );
+    afterEach(() => {
+      // Reset the object since it's a singleton
+      if (app._uninitialize) {
+        app._uninitialize();
+      }
     });
 
-    Object.values(FrameContexts).forEach(context => {
-      if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
-        it(`authentication.notifyFailure should successfully notify auth failure ${context} context`, async () => {
-          await utils.initializeWithContext('authentication');
+    describe('Testing authentication.initialize function', () => {
+      it('authentication.initialize should successfully register authentication.authenticate.success/failure handler', async () => {
+        authentication.initialize();
+        const messageForAuthenticationSuccess = framelessPostMock.findMessageByFunc(
+          'authentication.authenticate.success',
+        );
+        const messageForAuthenticationFailure = framelessPostMock.findMessageByFunc(
+          'authentication.authenticate.failure',
+        );
+        expect(messageForAuthenticationSuccess).toBeNull();
+        expect(messageForAuthenticationFailure).toBeNull();
+      });
+    });
 
-          authentication.notifyFailure(errorMessage);
+    describe('Testing authentication.registerAuthenticationHandlers function', () => {
+      Object.values(FrameContexts).forEach(context => {
+        if (!allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`authentication.registerAuthenticationHandlers should not allow calls from ${context} context`, async () => {
+            await framelessPostMock.initializeWithContext(context);
+            const authenticationParams: authentication.AuthenticatePopUpParameters = {
+              url: 'https://someurl/',
+              width: 100,
+              height: 200,
+            };
 
-          const message = utils.findMessageByFunc('authentication.authenticate.failure');
-          expect(message).not.toBeNull();
-          expect(message.args.length).toBe(1);
-          expect(message.args[0]).toBe(errorMessage);
+            expect(() => authentication.authenticate(authenticationParams)).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
```

Comment:
Shouldn't we have positive unit tests for `authentication.registerAuthenticationHandlers`? I only see tests for error cases.

Reply:
Added successful unit tests for frameless authentication.authenticate and authentication.registerAuthenticationHandlers.